### PR TITLE
AttributeFilter DisplayForm select component

### PR DESCRIPTION
**Pull request by ivanmjartan**
JIRA: RAIL-4420

---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9318/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11146/ : SUCCESS in 16m 55s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9338/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11170/ : SUCCESS in 16m 33s

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            className={buttonClassNames}
+            title={buttonTitle}
+            value={buttonTitle}
+            onClick={toggleDropdown}
+            iconRight={iconRight}
+        />
+    );
+};
+
+export const AttributeDisplayFormButton = injectIntl(WrappedAttributeDisplayFormButton);
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx:52]: Please replace this with `useIntl()` so we don't pollute the props with intl property + we can get rid of `Wrapped` component

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+export interface IToggleButtonProps {
+    text: string;
+    isOpened: boolean;
+    toggleDropdown: (e: React.SyntheticEvent) => void;
+}
+
+/**
+ * @internal
+ */
+export type ToggleButtonProps = IToggleButtonProps & WrappedComponentProps;
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx:20]: Naming seems a bit vague, what about `AttributeDisplayFormDropdownButton` + `IAttributeDisplayFormDropdownButtonProps` so we have it aligned with other components?

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdown.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    displayForms: IAttributeDisplayFormMetadataObject[];
+    selectedDisplayForm: ObjRef;
+    onSelect: (displayForm: ObjRef) => void;
+    alignPoints?: IAlignPoint[];
+}
+
+/**
+ * @internal
+ */
+export const AttributeDisplayFormDropdown: React.FC<IAttributeDisplayFormDropdownProps> = (props) => {
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdown.tsx:35]: Naming proposal: `AttributeDisplayFormSelect`

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownItem.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+export interface IDisplayFormDropdownItemProps {
+    displayForm: IAttributeDisplayFormMetadataObject;
+    onClick: (displayForm: ObjRef) => void;
+    selected: boolean;
+}
+
+/**
+ * @internal
+ */
+export const AttributeDisplayFormDropdownItem: React.FC<IDisplayFormDropdownItemProps> = (props) => {
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownItem.tsx:42]: Component name + props name should be aligned eg `AttributeDisplayFormSelectItem` `IAttributeDisplayFormSelectItemProps`

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeFilterDisplayFormSelect.tsx [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+/**
+ * @internal
+ */
+export const AttributeFilterDisplayFormSelect: React.FC<IAttributeFilterDisplayFormSelectProps> = (props) => {
+    const { onSelect, alignPoints } = props;
+
+    const { displayForms, currentDisplayFormRef } = useAttributeFilterContext();
+
+    if (displayForms.length <= 1) {
```
Comment (kandl) [inline libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeFilterDisplayFormSelect.tsx:24]: This condition seems like a something that should be in AD :/

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx (hunk 1 above)
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx:52]: done

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx (hunk 2 above)
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeDisplayFormDropdownButton.tsx:20]: done

File: libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeFilterDisplayFormSelect.tsx (hunk 5 above)
Comment (ivanmjartan) [reply libs/sdk-ui-filters/src/AttributeFilter@next/Components/Addons/AttributeDisplayFormSelect/AttributeFilterDisplayFormSelect.tsx:24]: Done

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9341/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11173/ : SUCCESS in 17m 07s

File: libs/sdk-ui-filters/src/AttributeFilter@next/hooks/useAttributeFilterControllerData.ts [hunk 6]
```diff
@@ -70,6 +71,8 @@ export function useAttributeFilterControllerData(
     const isFiltering = useIsFiltering(handler);
 
     const parentFilterAttributes = handler.getLimitingAttributeFiltersAttributes();
+    const displayForms = attribute ? attribute.displayForms : [];
```
Comment (no23reason) [inline libs/sdk-ui-filters/src/AttributeFilter@next/hooks/useAttributeFilterControllerData.ts:74]: Detail: should this be 
```
const displayForms = attribute?.displayForms ?? [];
```
that way you could avoid the
```
displayForms: displayForms ?? []
```
and just do `displayForms`

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build in pipeline gate aborted.

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2861

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2640/ : SUCCESS in 1m 18s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2941/ : SUCCESS in 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1774/ : SUCCESS in 18m 05s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11178/ : FAILURE in 17m 05s

**Comment by ivanmjartan** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/9347/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11181/ : SUCCESS in 16m 08s

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2861

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/2642/ : SUCCESS in 1m 25s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/2943/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/1776/ : SUCCESS in 31m 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/11185/ : SUCCESS in 15m 55s

**Pull request merged**